tests/cores.js

```diff
@@ -415,72 +416,94 @@ describe('Cores Methods', function() {
     })
   })
 
-  describe('submitHashAsync', () => {
+  describe('parse402Response', () => {
+    let lsat, challenge, response
     before(() => {
-      cores.setENV({ CHAINPOINT_CORE_CONNECT_IP_LIST: ['65.1.1.1'] })
-      cores.setRP(async () => {
-        throw 'No Invoice!'
-      })
+      challenge = data.challenge1000
+      lsat = Lsat.fromChallenge(challenge)
+      response = {
+        statusCode: 402,
+        headers: {
+          'WWW-Authenticate': challenge
+        }
+      }
     })
-    it('should return [] on 1 of 1 get invoice failure', async () => {
-      let result = await cores.submitHashAsync('deadbeefcafe')
-      expect(result).to.be.a('array')
-      expect(result.length).to.equal(0)
+
+    it('should throw if no LSAT challenge present in response or not a 402', () => {
+      const parseWrongStatusCode = () => cores.parse402Response({ ...response, statusCode: 401 })
+      const parseMissingHeader = () => cores.parse402Response({ statusCode: 402 })
+      expect(parseWrongStatusCode).to.throw()
+      expect(parseMissingHeader).to.throw()
+    })
+
+    it('should should return an LSAT with invoice information', () => {
+      const lsatFromResponse = cores.parse402Response(response)
+      expect(lsatFromResponse.invoice).to.exist
+      expect(lsatFromResponse.invoice).to.equal(lsat.invoice)
     })
   })
 
   describe('submitHashAsync', () => {
-    before(() => {
-      cores.setRP(async () => {
-        return { body: 'ok' }
-      })
-      cores.setENV({ MAX_SATOSHI_PER_HASH: 5, CHAINPOINT_CORE_CONNECT_IP_LIST: ['65.1.1.1'] })
+    let challengeResponse, env, coreList
```

Comment:
I organized these tests differently where there all `submitHashAsync` tests are grouped under a single describe. This had a few advantages when testing. The first is aesthetic, and so I admit purely subjective (so happy to convert to the other way if the other reasons don't convince you) where I think it's easier to read with one header followed by a bunch of `should` statements. The second is that it was easier to isolate tests so when I made an update I only wanted to see how the `submitHashAsync` tests behave and not all the tests in the page or library, then a single `describe.only` covers that. The third is that there were a lot of repeated variables that were easier to group in a single `beforeEach`. 

As I say, some of this is subjective but these changes did make dev/testing easier (also maybe subjective). If @jasonbukowski you think it'd be better the other way, let me know.

Reply:
If this way of organizing them worked better for you, and you feel it results in better formatted code and output, then I'm all for it.